subsys/net_core_monitor/app_core.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/types.h>
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <nrfx_ipc.h>
+#include "common.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(net_core_monitor, CONFIG_NET_CORE_MONITOR_LOG_LEVEL);
+
+#define IPC_MEM_CNT_IDX  0
+#define IPC_MEM_REAS_IDX 1
+
+int ncm_net_status_check(uint32_t * const reset_reas)
+{
+	volatile uint32_t gpmem;
```

Comment:
Why volatile? It's not a pointer to a peripheral. It's set/get by function after operations.

Reply:
It is unnecessary.